src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -986,10 +986,10 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     try {
                         X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslCallback.getSslSession().getPeerCertificates());
                         verifyEvidence(new X509PeerCertificateChainEvidence(x509Certificates));
```

Comment:
@darranl @dmlloyd , now verify evidence is delayed until an authorization request arrives, where the verification is only done in case state is other than name or authorized. Besides, certs for both peer and server are being cached separately.